### PR TITLE
Update cloudflare-turnstile types and CODEOWNERS

### PR DESCRIPTION
**Pull request by lbarthon**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://developers.cloudflare.com/turnstile/
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/cloudflare-turnstile/index.d.ts [hunk 1]
```diff
@@ -64,16 +115,37 @@ declare namespace Turnstile {
         /**
          * Optional. A JavaScript callback that is invoked when a challenge expires.
          */
-        "expired-callback"?: VoidFunction;
+        "expired-callback"?: (token: string) => void;
+
+        /**
+         * Optional. A JavaScript callback that is invoked when an error occurs with the widget.
+         */
+        "error-callback"?: VoidFunction | undefined;
```
Comment (punkeel) [inline types/cloudflare-turnstile/index.d.ts:123]: `| undefined` is redundant because of `?`, here and below?

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -949,7 +949,7 @@
 /types/cloud-env/                                         @Morfent
 /types/cloudflare/                                        @Xenfo
 /types/cloudflare-apps/                                   @MartynasZilinskas
-/types/cloudflare-turnstile/                              @jasongerbes
+/types/cloudflare-turnstile/                              @jasongerbes @lbarthon @worenga @olipayne @punkeel @migueldemoura
```
Comment (punkeel) [inline .github/CODEOWNERS:952]: isn't that file updated by the bot? (https://github.com/DefinitelyTyped/DefinitelyTyped?tab=readme-ov-file#definition-owners)

File: types/cloudflare-turnstile/index.d.ts (hunk 1 above)
Comment (lbarthon) [reply types/cloudflare-turnstile/index.d.ts:123]: It was done on all the previous types, I think it allows explicit undefined values, which can be helpful if the user passes a variable directly, that might be undefined

File: .github/CODEOWNERS (hunk 2 above)
Comment (lbarthon) [reply .github/CODEOWNERS:952]: oops

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/cloudflare-turnstile/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against da0da8173bad3d6d01f1ffa45ca9214f9b6e5870
</p>

**Comment by punkeel** [comment]
Just a friendly heads-up: the Turnstile product is under our ownership here at Cloudflare (see the shiny badge on our profile pages).

Huge thanks to @jasongerbes for your contributions so far! Feel free to let us know if you'd like to pass on your responsibilities.

**Review by punkeel — APPROVED** [review]

**Comment by jasongerbes** [comment]
Hey @punkeel, thanks for the context. I’m glad to see Cloudflare support these as the official types.

You’re welcome to remove me as a maintainer now that these types are being maintained by Cloudflare.

**Comment by jasongerbes** [comment]
@punkeel , while it’s certainly not my call, could I suggest adding a reference to the `@types/cloudflare-turnstile` package to the Turnstile Docs.

Perhaps the [Client-side rendering](https://developers.cloudflare.com/turnstile/get-started/client-side-rendering) page, for integrators using TypeScript to render the Turnstile widget.

**Comment by lbarthon** [comment]
Hey @jasongerbes! Really appreciate you taking care of that beforehand, I think we can take it from here and remove you from the maintainers, as that seems to be fine for you.
Once this is updated and we made all the changes we want, we will for sure mention it in the Cloudflare documentation!

**Review by jasongerbes — APPROVED** [review]

**Comment by lbarthon** [comment]
Ready to merge

**Pull request merged**